src/main/java/emu/grasscutter/game/managers/blossom/BlossomManager.java

```diff
@@ -0,0 +1,237 @@
+        var id = gadget.getGadgetId();
+        if(BlossomType.valueOf(id)==null){
+            return;
+        }
+        gadget.buildContent();
+        gadget.setState(204);
+        int worldLevel = getWorldLevel();
+        GadgetWorktop gadgetWorktop = ((GadgetWorktop) gadget.getContent());
+        gadgetWorktop.addWorktopOptions(new int[]{187});
+        gadgetWorktop.setOnSelectWorktopOptionEvent((GadgetWorktop context, int option) -> {
+            BlossomActivity activity;
+            EntityGadget entityGadget = context.getGadget();
+            synchronized (blossomActivities) {
+                for (BlossomActivity activeLeyLines : this.blossomActivities) {
+                    if (activeLeyLines.getGadget() == entityGadget) {
+                        return false;
+                    }
+                }
+
+                int volume=0;
+                IntList monsters = new IntArrayList();
+                while(true){
+                    var remain = GameDepot.getBlossomConfig().getMonsterFightingVolume() - volume;
+                    if(remain<=0){
+                        break;
+                    }
+                    var rand = Utils.randomRange(1,100);
+                    if(rand>85 && remain>=50){//15% ,generate strong monster
+                        monsters.addAll(getRandomMonstersID(2,1));
+                        volume+=50;
+                    }else if(rand>50 && remain>=20) {//35% ,generate normal monster
+                        monsters.addAll(getRandomMonstersID(1,1));
+                        volume+=20;
+                    }else{//50% ,generate weak monster
+                        monsters.addAll(getRandomMonstersID(0,1));
+                        volume+=10;
+                    }
+                }
+
+                Grasscutter.getLogger().info("Blossom Monsters:"+monsters);
+
+                activity = new BlossomActivity(entityGadget, monsters, -1, worldLevel);
+                blossomActivities.add(activity);
+            }
+            entityGadget.updateState(201);
+            scene.setChallenge(activity.getChallenge());
+            scene.removeEntity(entityGadget, VisionTypeOuterClass.VisionType.VISION_TYPE_REMOVE);
+            activity.start();
+            return true;
+        });
+        createdEntity.add(gadget);
+        notifyIcon();
+    }
+
+    public void notifyIcon() {
+        final int wl = getWorldLevel();
+        final int worldLevel = (wl < 0) ? 0 : ((wl > 8) ? 8 : wl);
+        final int monsterLevel = GameData.getWorldLevelDataMap().get(worldLevel).getMonsterLevel();
+        List<BlossomBriefInfoOuterClass.BlossomBriefInfo> blossoms = new ArrayList<>();
+        GameDepot.getSpawnLists().forEach((gridBlockId, spawnDataEntryList) -> {
+            int sceneId = gridBlockId.getSceneId();
+            spawnDataEntryList.stream()
+                .map(SpawnDataEntry::getGroup)
+                .map(SpawnGroupEntry::getSpawns)
+                .flatMap(List::stream)
```

Comment:
This feels like it's going around in circles but I can't be bothered to confirm if it's safe to just delete this and use the raw `SpawnDataEntry` stream.